### PR TITLE
rhv provider for reference deployments

### PR DESCRIPTION
**Pull request by cooktheryan**
Will need to be cherry picked for 3.10/3.11

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@cooktheryan - Hey Ryan, does this need to go into master, which will publish this on docs.okd.io? https://github.com/openshift/openshift-docs/pull/12043 which is OpenShift on OpenStack is only going in 3.10.

I don't know the correct answer but want to make sure we are consistent with this so either way is fine with me

@xltian - can you please assign someone to review it?

@ahardin-rh - can you please take a look once we have a QE review?

**Comment by cooktheryan** [comment]
@vikram-redhat similar to what was done with #11670 if that helps

**Comment by xltian** [comment]
@xingxingxia is working on reviewing it.

File: install_config/topics/full_rhv_inventory.adoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+openshift_node_debug_level="{{ node_debug_level | default(debug_level, true) }}"
+openshift_enable_service_catalog=False
+
+app_dns_prefix=apps
+public_hosted_zone=example.com
+load_balancer_hostname=lb.{{public_hosted_zone}}
+openshift_master_cluster_hostname="{{ load_balancer_hostname }}"
+openshift_master_cluster_public_hostname=openshift-master.{{ public_hosted_zone }}
+openshift_master_default_subdomain="{{ app_dns_prefix }}.{{ public_hosted_zone }}"
+openshift_public_hostname="{{openshift_master_cluster_public_hostname}}"
```
Comment (xingxingxia) [inline install_config/topics/full_rhv_inventory.adoc]: openshift_master_cluster_public_hostname could be just as same as "{{ load_balancer_hostname }}"
The line openshift_public_hostname=... here could be removed

File: install_config/topics/full_rhv_inventory.adoc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+# Pod Networking
+os_sdn_network_plugin_name=redhat/openshift-ovs-networkpolicy
+
+# Registry
+openshift_hosted_registry_storage_kind=glusterfs
+
+# Authentication (example here creates one user, myuser with password changeme)
+openshift_master_identity_providers="[{'name': 'htpasswd_auth', 'login': 'True', 'challenge': 'True', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]"
```
Comment (xingxingxia) [inline install_config/topics/full_rhv_inventory.adoc]: > 'login': 'True', 'challenge': 'True'

Use lower case `true`

File: install_config/topics/full_rhv_inventory.adoc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+openshift_master_htpasswd_users={'myuser': '$apr1$zAhyA9Ko$rBxBOwAwwtRuuaw8OtCwH0'}
+
+# Red Hat Subscription Management
+# (vault_ variables are meant to be provided by Ansible Vault to protect passwords)
+rhsub_pool=Red Hat OpenShift Container Platform*
+rhsub_user="{{ vault_rhsub_user }}"
+rhsub_password="{{ vault_rhsub_password }}"
```
Comment (xingxingxia) [inline install_config/topics/full_rhv_inventory.adoc]: The lines for subscription are not needed in inventory file IMO

File: install_config/topics/full_rhv_inventory.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+[lb]
+lb.example.com
+
+[nodes]
+master0.example.com openshift_node_group_name=node-config-master
+master1.example.com openshift_node_group_name=node-config-master
+master2.example.com openshift_node_group_name=node-config-master
+infra0.example.com openshift_node_group_name=node-config-infra glusterfs_devices="['/dev/vdd']"
+infra1.example.com openshift_node_group_name=node-config-infra glusterfs_devices="['/dev/vdd']"
+infra2.example.com openshift_node_group_name=node-config-infra glusterfs_devices="['/dev/vdd']"
```
Comment (xingxingxia) [inline install_config/topics/full_rhv_inventory.adoc]: The `glusterfs_devices=...` should be put under  `[glusterfs_registry]` section, see the https://docs.openshift.com/container-platform/3.10/install_config/persistent_storage/persistent_storage_glusterfs.html#install-example-registry

File: install_config/topics/full_rhv_inventory.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+openshift_node_local_quota_per_fsgroup=512Mi
+openshift_use_system_containers=False
+
+[masters:vars]
+# Use the container_runtime section of the prerequisites playbook to set up extra file systems
+container_runtime_extra_storage="[{'device': '/dev/vdc', 'path': '/var/lib/origin/openshift.local.volumes', 'options': 'gquota', 'filesystem': 'xfs', 'format': 'True'}, {'device': '/dev/vdd', 'path': '/var/lib/etcd', 'hosts': 'masters', 'filesystem': 'xfs', 'format': 'True'}]"
+
+[nodes:vars]
+# Use the container_runtime section of the prerequisites playbook to set up extra file systems
+container_runtime_extra_storage="[{'device': '/dev/vdc', 'path': '/var/lib/origin/openshift.local.volumes', 'options': 'gquota', 'filesystem': 'xfs', 'format': 'True'}]"
```
Comment (xingxingxia) [inline install_config/topics/full_rhv_inventory.adoc]: I'm not quite sure if above `container_runtime_extra_storage=...` has no problem, @jianlinliu could you help confirm?

Comment (jianlinliu) [reply install_config/topics/full_rhv_inventory.adoc]: I have no idea about the option neither.

Comment (cooktheryan) [reply install_config/topics/full_rhv_inventory.adoc]: https://github.com/openshift/openshift-ansible/blob/master/inventory/hosts.example#L146 It allows for partitions to be created, mounted, and added to fstab to be used later on to segment off filesystems. It's important for the openshift_node_local_quota_per_fsgroup on all of the nodes. But its just an optional setting if we wanted we could remove it and make some inventory changes. It's just a nice to have

File: install_config/topics/full_rhv_inventory.adoc (hunk 2 above)
Comment (xingxingxia) [reply install_config/topics/full_rhv_inventory.adoc]: > 'challenge': 'True'

Both use lower case `true`

**Comment by xingxingxia** [comment]
1 configuring_rhv.adoc seems to not deliver much useful info, maybe it could be not needed?
2 https://github.com/openshift/openshift-docs/pull/12067#discussion_r219106085
Except above comments, the doc looks good from my side

**Comment by cooktheryan** [comment]
@xingxingxia the reason why configuring_rhv.adoc is that way is to provide modularity when changes and updates come in 3.11 for example the flex driver.  A second can just be added to topics. This similar format was used for azure and gcp

**Comment by cooktheryan** [comment]
@xingxingxia @ahardin-rh this has been squashed due to the request above

**Comment by cooktheryan** [comment]
@vikram-redhat can this be assigned. The issues above have been resolved and this has been squashed

**Comment by vikram-redhat** [comment]
@cooktheryan thanks. 

@openshift/team-documentation - PTAL for peer review.

**Comment by kalexand-rh** [comment]
@vikram-redhat, where did you want this to go in the TOC?

**Comment by vikram-redhat** [comment]
@cooktheryan Ryan - was the location of these docs discussed with @ahardin-rh or @aheslin ?

**Comment by cooktheryan** [comment]
This should match the VMware, GCP, and Azure providers in location and format

**Comment by kalexand-rh** [comment]
Perfect. I'm going to merge this change now and apply some style edits and wire it in tomorrow. Thanks!

**Comment by vikram-redhat** [comment]
@kalexand-rh  - was there a follow up to this?

**Pull request merged**